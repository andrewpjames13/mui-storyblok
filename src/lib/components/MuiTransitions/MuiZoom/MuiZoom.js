import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import { Zoom } from '@material-ui/core';
import StoryBlok from '../../../utils/Storyblok';
import MuiGridList from '../../MuiGridList/MuiGridList';

const MuiZoom = ({
  rootClass,
  content,
  zoomIn,
  timeout,
}) => {
  const components = {
    MuiGridList,
  };

  const styles = StoryBlok.arrayToMuiStyles(rootClass);

  return (
    <div className={styles.root}>
      <Zoom
        in={zoomIn}
        timeout={timeout}
      >
        <div>
          {content.map((item, index) => createElement(
            components[item.component],
            Object.assign(item, { key: index }),
          ))}
        </div>
      </Zoom>
    </div>
  );
};

export default MuiZoom;

MuiZoom.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui props: bool
   * If true, the component will transition in.
   */
  zoomIn: PropTypes.bool,
  /**
   * mui props: number | string
   * The duration for the transition, in milliseconds.
   * You may specify a single timeout for all transitions, or individually with an object.
   * Set to 'auto' to automatically calculate transition time based on height.
   */
  timeout: PropTypes.string,
  /**
   * Content passed to render
   * components: MuiGridList
   */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiZoom.defaultProps = {
  zoomIn: true,
  timeout: 'auto',
  rootClass: [],
};
