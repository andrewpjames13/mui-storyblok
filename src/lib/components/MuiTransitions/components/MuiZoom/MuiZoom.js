import React from 'react';
import PropTypes from 'prop-types';
import { Zoom } from '@material-ui/core';
import StoryBlok from '../../../../utils/Storyblok';
import Blok from '../../../StoryBlokPage/components/Blok/Blok';

const MuiZoom = ({
  rootClass,
  content,
  zoomIn,
  enter,
  exit,
  timeout,
}) => {
  const styles = StoryBlok.arrayToMuiStyles(rootClass);

  return (
    <div className={styles.root}>
      <Zoom
        in={zoomIn}
        timeout={timeout}
      >
        <div>
          {content.map((item, index) => (
            <Blok {...item} key={index} />
          ))}
        </div>
      </Zoom>
    </div>
  );
};

export default MuiZoom;

MuiZoom.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui props: bool
   * If true, the component will transition in.
   */
  zoomIn: PropTypes.bool,
  /**
   * mui props: number | string
   * The duration for the transition, in milliseconds.
   * You may specify a single timeout for all transitions, or individually with an object.
   * Set to 'auto' to automatically calculate transition time based on height.
   */
  timeout: PropTypes.string,
  /**
   * mui props: number
   * Number of milliseconds for component to effect on screen.
   */
  enter: PropTypes.string,
  /**
   * mui props: number
   * Number of milliseconds for component to effect off screen.
   */
  exit: PropTypes.string,
  /**
   * Content passed to render
   * components: Blok
   */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiZoom.defaultProps = {
  zoomIn: true,
  timeout: 'auto',
  enter: 1000,
  exit: 1000,
  rootClass: [],
};
