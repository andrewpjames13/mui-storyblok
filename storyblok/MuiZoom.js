module.exports = {
  "name": "MuiZoom",
  "display_name": null,
  "updated_at": "2020-05-13T18:41:57.250Z",
  "schema": {
    "rootClass": {
      "type": "options",
      "description": "* stroyblok multiselect of css classes\n* Mui Override or extend the styles applied to the component."
    },
    "content": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "Blok"
      ],
      "required": true,
      "maximum": "1"
    },
    "zoomIn": {
      "type": "number",
      "max_length": "",
      "description": "* mui props: number\n* Number of milliseconds for component to transition in.",
      "default_value": "300"
    },
    "enter": {
      "type": "number",
      "description": "* mui props: number\n* Number of milliseconds for component to effect on screen.",
      "display_name": "",
      "default_value": "1000"
    },
    "exit": {
      "type": "number",
      "description": "* mui props: number\n* Number of milliseconds for component to effect off screen.",
      "default_value": "1000"
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiZoom",
  "component_group_uuid": null
}