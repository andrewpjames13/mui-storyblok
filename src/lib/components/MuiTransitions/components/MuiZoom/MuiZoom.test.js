import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiZoom from './MuiZoom';

function setup() {
  const props = {
    enter: 1500,
    exit: 1500,
    content: [{
      component: 'Blok',
      content: [{
        component: 'MuiGrid',
        content: [{
          component: 'MuiGridItem',
          content: [{
            component: 'MuiCard',
            content: [{
              component: 'MuiCardContent',
              content: [{
                component: 'MuiTypography',
                content: [{
                  component: 'MuiText',
                  text: 'Juice Box',
                }],
              }],
            }],
          }],
        }],
      }],
    }],
  };

  const comp = shallow(<MuiZoom {...props} />);
  return { comp, props };
}

describe('<MuiZoom />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(
      <MemoryRouter>
        <MuiZoom {...props} />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render MuiGridList', () => {
    const { comp } = setup();
    expect(comp).toMatchSnapshot();
  });
});
