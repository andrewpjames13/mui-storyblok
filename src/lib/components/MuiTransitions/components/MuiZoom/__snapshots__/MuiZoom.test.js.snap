// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiZoom /> should render MuiGridList 1`] = `ShallowWrapper {}`;

exports[`<MuiZoom /> snapshot 1`] = `
<div
  className="makeStyles-root-2"
>
  <div
    style={
      Object {
        "transform": "scale(0)",
        "visibility": "hidden",
      }
    }
  />
</div>
`;
